pkgs/tools/text/mdbook/default.nix

```diff
@@ -15,15 +15,17 @@ rustPlatform.buildRustPackage rec {
 
   buildInputs = stdenv.lib.optionals stdenv.isDarwin [ CoreServices ];
 
+  # FIXME: Use impure version of CoreFoundation because of missing symbols.
+  #   Undefined symbols for architecture x86_64: "_CFURLResourceIsReachable"
```

Comment:
I think darwin.cf-private does this for you

Reply:
Actually not sure if that resolves it

Reply:
That's what https://discourse.nixos.org/t/help-with-rust-linker-error-on-darwin-cfurlresourceisreachable/2657 tried to use and it doesn't work there.

I _think_ what's happening there is the CFLAGS references both the cf-private framework which is a symlink to the real one, _and_ the Nix-built framework which is missing the symbols, and the linker picks the Nix-built one as the path to resolve the framework reference too. But I'm just speculating at this point.

Reply:
I just dug into this and the store path for `darwin.cf-private` actually contains a Nix-built copy of CoreFoundation instead of a symlink to the system CoreFoundation.

```
> exa --tree (nix path-info nixpkgs.darwin.cf-private)
/nix/store/28j21gvjwmsrwyh7xn60h4fmwqs5b5fi-swift-corefoundation-private/
├── Library/
│  └── Frameworks/
│     └── CoreFoundation.framework/
│        ├── CoreFoundation -> Versions/Current/CoreFoundation
│        ├── Headers -> Versions/Current/Headers
│        ├── Modules -> Versions/Current/Modules
│        └── Versions/
│           ├── A/
│           │  ├── CoreFoundation*
│           │  ├── Headers/
│           │  │  ├── …
│           │  ├── Modules/
│           │  │  └── module.modulemap
│           │  └── PrivateHeaders/
│           └── Current -> A
└── nix-support/
   └── setup-hook
```

Reply:
Looks like `cf-private` is supposed to contain that, but adds a setup hook that stuffs the real Foundation into `NIX_LDFLAGS`.

Given that, I tried actually building the derivation with `cf-private`, and it worked! I can't figure out how to get this to work in `nix-shell`, I guess setup hooks aren't being executed that way? But if it works in the derivation, that's good enough.

I'll update this PR, and also update `watchexec` to match.